forwarder.go

```diff
@@ -226,28 +227,76 @@ func buildExternalURL(newTalariaName, domain string) string {
  * Extracts IP address, certificate provider, and expiry details from HTTP request headers.
  * Calls the RI API with this information.
  */
-func updateResourceIpAddressAndCertificateInfo(req *http.Request, client *http.Client, resourceURL *url.URL) error {
 
-	certificateProviderRaw := req.Header.Get(certificateProviderHeader)
+// base64Decode decodes a base64 encoded string.
+func base64Decode(encodedStr string) ([]byte, error) {
+	decodedData, err := base64.StdEncoding.DecodeString(encodedStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode base64 string: %v", err)
+	}
+	return decodedData, nil
+}
+
+// extractWebPAConveyHeader extracts the X-WebPA-Convey header value.
+func extractWebPAConveyHeader(req *http.Request) (string, error) {
+	xWebPACode := req.Header.Get("X-WebPA-Convey")
```

Comment:
If convey header is empty, do we need to throw an error here?
In current scenario, if convey headers are not coming by any chance, CPE will not be able to connect to backend as petasos rewrite will return an error.

Reply:
Instead of throwing an error when X-WebPA-Convey is absent, if the X-WebPA-Convey field is present, we will simply extract the resources and update them accordingly. If the field is not present, the flow will remain unchanged, as it was previously.